api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java

```diff
@@ -440,61 +479,185 @@ public void testCreateCohortBadCriteria() {
   }
 
   @Test
-  public void testUpdateCohort() throws Exception {
-    Cohort cohort = createDefaultCohort();
-    cohort =
+  public void testUpdateCohortAccessLevelOwner() {
+    Cohort saved = createTestCohort(WorkspaceAccessLevel.WRITER);
+    // update
+    Cohort toUpdate = saved.name(UPDATED_COHORT_NAME);
+    CLOCK.increment(1000); // lets say time ticked 1 sec past
+    cohortsController
+        .updateCohort(workspace.getNamespace(), workspace.getId(), toUpdate.getId(), toUpdate)
+        .getBody();
+    Cohort updated =
         cohortsController
-            .createCohort(workspace.getNamespace(), workspace.getId(), cohort)
+            .getCohort(workspace.getNamespace(), workspace.getId(), saved.getId())
             .getBody();
+    assertThat(updated.getName()).isEqualTo(UPDATED_COHORT_NAME);
+    assertThat(updated.getCriteria()).isEqualTo(saved.getCriteria());
+    assertThat(updated.getCreationTime()).isEqualTo(saved.getCreationTime());
+    assertThat(updated.getLastModifiedTime()).isGreaterThan(saved.getLastModifiedTime());
+  }
 
-    cohort.setName("updated-name");
+  @Test
+  public void testUpdateCohortAccessLevelWriter() {
+    Cohort saved = createTestCohort(WorkspaceAccessLevel.WRITER);
+    // update
+    stubGetWorkspace(
+        workspace.getNamespace(), workspace.getName(), CREATOR_EMAIL, WorkspaceAccessLevel.WRITER);
+    stubGetWorkspaceAcl(
+        workspace.getNamespace(), workspace.getName(), CREATOR_EMAIL, WorkspaceAccessLevel.WRITER);
+    Cohort toUpdate = saved.name(UPDATED_COHORT_NAME);
+    CLOCK.increment(1000); // lets say time ticked 1 sec past
+    cohortsController
+        .updateCohort(workspace.getNamespace(), workspace.getId(), toUpdate.getId(), toUpdate)
+        .getBody();
     Cohort updated =
         cohortsController
-            .updateCohort(workspace.getNamespace(), workspace.getId(), cohort.getId(), cohort)
+            .getCohort(workspace.getNamespace(), workspace.getId(), saved.getId())
             .getBody();
```

Comment:
Unnecessary call. cohortsController#updateCohort will return the updated cohort.

Reply:
updated code and removed this call